specification/containerservice/resource-manager/Microsoft.ContainerService/preview/2022-10-02-preview/managedClusters.json

```diff
@@ -5113,6 +5113,39 @@
           },
           "title": "The upgrade channel for auto upgrade. The default is 'none'.",
           "description": "For more information see [setting the AKS cluster auto-upgrade channel](https://docs.microsoft.com/azure/aks/upgrade-cluster#set-auto-upgrade-channel)."
+        },
+        "nodeOSUpgradeChannel": {
+          "type": "string",
+          "enum": [
+            "Unmanaged",
+            "None",
+            "SecurityPatch",
+            "NodeImage",
+          ],
+          "x-ms-enum": {
+            "name": "nodeOSUpgradeChannel",
+            "modelAsString": true,
+            "values": [
+              {
+                "value": "None",
+                "description": "No attempt to update your machines OS will be made either by OS or by rolling VHDs. This means you are responsible for your security updates"
+              },
+              {
+                "value": "Unmanaged",
+                "description": "OS updates will be applied automatically through the OS built-in patching infrastructure. Newly scaled in machines will be unpatched initially and will be patched at some point by the OS's infrastructure. Behavior of this option depends on the OS in question. Ubuntu and Mariner apply security patches through unattended upgrade roughly once a day around 06:00 UTC. Windows does not apply security patches automatically and so for them this option is equivalent to None till further notice"
+              },
+              {
+                "value": "SecurityPatch",
+                "description": "AKS will update the nodes VHD with patches from the image maintainer labelled "security only" on a regular basis. Where possible, patches will also be applied without reimaging to existing nodes. Some patches, such as kernel patches, cannot be applied to existing nodes without disruption. For such patches, the VHD will be updated, and machines will be rolling reimaged to that VHD following maintenance windows and surge settings. This option incurs the extra cost of hosting the VHDs in your node resource group."
+              },
+              {
+                "value": "NodeImage",
+                "description": "AKS will update the nodes with a newly patched VHD containing security fixes and bugfixes on a weekly cadence. With the VHD update machines will be rolling reimaged to that VHD following maintenance windows and surge settings. No extra VHD cost is incurred when choosing this option as AKS hosts the images."
+              }
+            ]
+          },
+          "title": "Node OS Upgrade Channel",
```

Comment:
Actually thinking this title should say: "Manner in which the OS on your nodes is updated."
and then the description can say: "The default is Unmanaged, but may change to either NodeImage or SecurityPatch at GA."

Since the current title isn't really saying much (it's just repeating the name of the property)

Reply:
this makes sense if the property name is always shown with the title. If sometimes just the title is shown it's worse checking.....

Reply:
@matthchr  title and descriotion is confusing me

https://learn.microsoft.com/en-us/rest/api/aks/managed-clusters/get?tabs=HTTP#managedclusterautoupgradeprofile

ManagedClusterAutoUpgradeProfile only has a description not a title
upgrade channel shows the description only when looking at upgrade profile in the but shows the title in you click on it but not the description.... :)